### PR TITLE
Message Structure and Protocol URIs

### PR DESCRIPTION
**Pull request by TelegramSam**
This work is a partial application of the DIDComm profile of jwm (also in the repo) and some included details about protocol type URIs that related.
Signed-off-by: Sam Curren <telegramsam@gmail.com>

### CODE REVIEW

File: message_structure.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+```
+
+
+
+The attributes of a DIDComm Message are as follows:
+
+- **id** - REQUIRED. Message ID. The `id` attribute value MUST be unique to the sender.
+- **type** - REQUIRED. Message Type. The `type` attribute value MUST be a valid [MTURI](https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0003-protocols/uris.md#mturi), that when resolved gives human readable information about the message. The attributes value also informs the content of the message, for example the presence of other attributes and how they should be processed.
+- **to** - OPTIONAL. Recipient(s) identifier. The `to` attribute MUST be an array of strings where each element is a valid [DID](https://w3c.github.io/did-core/#generic-did-syntax) which identifies the recipients of the message.
+- **from** - OPTIONAL. Sender identifier. The `from` attribute MUST be a string that is a valid [DID](https://w3c.github.io/did-core/#generic-did-syntax) which identifies the sender of the message. For DID methods that use query parameters to carry additional information, they might also be present in the from string.
```
Comment (kdenhartog) [inline message_structure.md:28]: I think we have to require this now to get did linkage because it's not coming in the outer envelope anymore.

Comment (TelegramSam) [reply message_structure.md:28]: I was going to address this in the section about encryption, but I agree that a mention here is also useful.

**Comment by TelegramSam** [comment]
Discussed in WG Call 2020-05-11. Updated to attribute to be a list instead of a singular string.

**Comment by awoie** [comment]
@TelegramSam this PR seems uncontroversial, I think we can merge this.

File: message_structure.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	}
+}
+```
+
+
+
+The attributes of a DIDComm Message are as follows:
+
+- **id** - REQUIRED. Message ID. The `id` attribute value MUST be unique to the sender.
+- **type** - REQUIRED. Message Type. The `type` attribute value MUST be a valid [Message Type URI](protocols.md#message-type-uri) , that when resolved gives human readable information about the message. The attributes value also informs the content of the message, for example the presence of other attributes and how they should be processed.
```
Comment (peacekeeper) [inline message_structure.md:26]:
```suggestion
- **@id** - REQUIRED. Message ID. The `@id` attribute value MUST be unique to the sender.
- **@type** - REQUIRED. Message Type. The `@type` attribute value MUST be a valid [Message Type URI](protocols.md#message-type-uri) , that when resolved gives human readable information about the message. The attributes value also informs the content of the message, for example the presence of other attributes and how they should be processed.
```

Comment (TelegramSam) [reply message_structure.md:26]: Good catch! I believe this should be updated the other way - by removing the `@` prefix in the examples. This does have implications about JSON-LD compatibility that we have not discussed. I'll put this specific topic on the agenda for Monday's meeting and we can update appropriately to the decisions there.

**Pull request merged**